docs/ossm/ambient/waypoint.md

```diff
@@ -0,0 +1,229 @@
+
+* **Traffic Management:** This includes advanced HTTP routing and load balancing, circuit breaking, rate limiting, fault injection, retries, and timeouts.
+* **Security:** For rich authorization policies based on L7 primitives like request type or HTTP headers.
+* **Observability:** To collect HTTP metrics, enable access logging, and implement tracing for your applications.
+
+**Trade-offs:**
+
+* Ambient mode is a newer architecture and may have different operational considerations compared to the traditional sidecar model.
+* L7 features require the deployment of `waypoint` proxies, which add a small amount of overhead for the services that utilize them.
+
+---
+
+## 2. Pre-requisites to Using Waypoint Ambient Mode with OSSM 3
+
+Before installing Istio's ambient mode with OpenShift Service Mesh, ensure the following prerequisites are met:
+
+* **OpenShift Container Platform 4.19+:** This version of OpenShift is required for supported Kubernetes Gateway API CRDs, which are essential for ambient mode functionalities.
+* **OpenShift Service Mesh 3.1.0+ operator is installed:** Ensure that the OSSM operator version 3.1.0 or later is installed on your OpenShift cluster.
+* **Istio deployed in Ambient Mode:** Refer to OSSM Ambient [initial doc](README.md).
+* **Labeling:** Make sure the workloads/namespaces have the appropriate labels for ztunnel traffic redirection.
+
+**Pre-existing Service Mesh Installations:**
+
+While the use of properly defined [discovery selectors](https://istio.io/latest/docs/ops/configuration/mesh/configuration-scoping/#discoveryselectors) will allow a service mesh to be deployed in ambient mode alongside a service mesh in sidecar mode, this is not a scenario we have thoroughly validated. To avoid potential conflicts, as a technology preview feature, Istio's ambient mode should only be installed on clusters without a pre-existing OpenShift Service Mesh installation.
+
+**Note**: Istio's ambient mode is completely incompatible with clusters containing the OpenShift Service Mesh 2.6 or earlier versions of the operator and they should not be used together.
+
+---
+
+## 3. Waypoint Proxies
+
+### 3.1 Using Kubernetes Gateway API
+
+Waypoint proxies are deployed using Kubernetes Gateway resources.  
+
+As of OpenShift 4.19, the Kubernetes Gateway API CRDs comes pre-installed and are supported as part of OpenShift Container Platform.
+
+**Note:** As of OpenShift 4.17, the Kubernetes Gateway API CRDs are not available by default and must be installed to be used. This can be done with the following command:
+
+```bash
+oc get crd gateways.gateway.networking.k8s.io &> /dev/null || \
+  { oc apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.2.0/standard-install.yaml; }
+```
+
+### 3.2 Set up Istio Ambient Mode Resources and a Sample Application
+
+1. Install the Sail Operator along with Istio in ambient mode using the following [steps](README.md#3-procedure-to-install-istios-ambient-mode). 
```

Comment:
```suggestion
1. If you have not already done so, install the OpenShift Service Mesh Operator along with Istio in ambient mode using the following [steps](README.md#3-procedure-to-install-istios-ambient-mode). 
```
This is a pre-requisite, so technically doesn't need to be repeated... but we can leave it for completeness.

Reply:
Updated.